debian/mariadb-common.postrm

```diff
@@ -1,9 +1,16 @@
-#!/bin/bash
+#!/bin/sh
+
 set -e
 
-if [ "$1" = "purge" ]; then
-  rmdir /etc/mysql/conf.d 2>/dev/null || true
-  rmdir /etc/mysql 2>/dev/null || true
-fi
+case "$1" in
+  remove|disappear)
+    # New packaging paradigm for my.cnf handling among MySQL variants
+    # Used in Ubuntu since Dec-2014 and in Debian since Jul-2015
+    if [ -f /usr/share/mysql-common/configure-symlinks ]
+    then
+      /usr/share/mysql-common/configure-symlinks remove mariadb "/etc/mysql/mariadb.cnf"
+    fi
+  ;;
+esac
```

Comment:
no "purge" anymore?

Reply:
Tested on an installed system that `/var/lib/dpkg/info/mariadb-common.postrm` does not get any automatic additions form debhelper and remains just as it is in the packaging source.

For mariadb.org packages this does not even matter, since our mysql-common is installed and there is no `/usr/share/mysql-common/configure-symlinks` on the system anyway, so not even the remove runs.

Reply:
I also tested that purge of mariadb-common removes correct files, as does a purge of mysql-common. There is no need to do any special purge operations in the postinst file.